environment/integration/helpers.go

```diff
@@ -34,6 +36,40 @@
 	})))
 }
 
+// createTestTempDir creates a temporary directory with a shorter path on Windows
+func createTestTempDir(t *testing.T, prefix string) string {
+	var dir string
+	var err error
+
+	// Get test name and replace slashes with underscores to avoid path separator issues
+	testName := t.Name()
+	testName = strings.ReplaceAll(testName, "/", "_")
+
+	if runtime.GOOS == "windows" {
+		// On Windows, use a shorter base path to avoid path length issues
+		// Generate a unique short name using last 8 chars of test name
+		if len(testName) > 8 {
+			testName = testName[len(testName)-8:]
+		}
+		// Use C:\Temp or system temp root
+		tempRoot := filepath.Join(filepath.VolumeName(os.TempDir()), "\\Temp")
+		if err := os.MkdirAll(tempRoot, 0755); err != nil {
+			// Fall back to regular temp dir if we can't create C:\Temp
+			tempRoot = os.TempDir()
+		}
+		dir, err = os.MkdirTemp(tempRoot, prefix+testName+"-*")
```

Comment:
🔧 **Maintainability (Medium):** Remove double backslash in Windows path construction - filepath.Join will handle path separators correctly

```suggestion
	if runtime.GOOS == "windows" {
		// On Windows, use a shorter base path to avoid path length issues
		// Generate a unique short name using last 8 chars of test name
		if len(testName) > 8 {
			testName = testName[len(testName)-8:]
		}
		// Use C:\Temp or system temp root
		tempRoot := filepath.Join(filepath.VolumeName(os.TempDir()), "Temp")
		if err := os.MkdirAll(tempRoot, 0755); err != nil {
			// Fall back to regular temp dir if we can't create C:\Temp
			tempRoot = os.TempDir()
		}
```